sycl/include/sycl/ext/oneapi/bfloat16.hpp

```diff
@@ -178,11 +181,13 @@ class bfloat16 {
     return type{static_cast<float>(lhs) op static_cast<float>(rhs)};           \
   }                                                                            \
   template <typename T>                                                        \
-  friend type operator op(const bfloat16 &lhs, const T &rhs) {                 \
+  friend std::enable_if_t<std::is_convertible_v<T, float>, type> operator op(  \
```

Comment:
Not about this PR, but rather about the approach used for `bfloat16`. IMO this line is one more proof of using wrong approach for `bfloat16` comparing to `sycl::half`.

Look, here the return type of operator+ is always `bfloat16`, while it should be determined by C++ type-promotion-rules usually choosing a heavier type, e.g. 'double` + `float` results into `double`.

Here: `double` + `bfloat16` is `bfloat16`, which is wrong IMO.